app/build.gradle.kts

```diff
@@ -0,0 +1,173 @@
+import java.util.Properties
+import java.io.FileInputStream
+
+plugins {
+    id("com.android.application")
+    id("org.jetbrains.kotlin.android")
+    id("dagger.hilt.android.plugin")
+    id("com.google.dagger.hilt.android")
+    id("com.mikepenz.aboutlibraries.plugin")
+    id("com.google.devtools.ksp")
+}
+
+fun fetchGitCommitHash(): String {
+    val process = ProcessBuilder("git", "rev-parse", "--verify", "--short", "HEAD")
+        .redirectErrorStream(true)
+        .start()
+    return process.inputStream.bufferedReader().use { it.readText().trim() }
+}
+
+val gitCommitHash = fetchGitCommitHash()
+val keyProps = Properties()
+val keyPropsFile = rootProject.file("signature/keystore.properties")
+if (keyPropsFile.exists()) {
+    println("Loading keystore properties from ${keyPropsFile.absolutePath}")
+    keyProps.load(FileInputStream(keyPropsFile))
+}
+
+android {
+    compileSdk = 34
+
+    defaultConfig {
+        applicationId = "me.ash.reader"
+        minSdk = 26
+        targetSdk = 33
+        versionCode = 25
+        versionName = "0.10.1"
+
+        buildConfigField("String", "USER_AGENT_STRING", "\"ReadYou/${'$'}{versionName}(${versionCode})\"")
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+        vectorDrawables.useSupportLibrary = true
+
+        javaCompileOptions {
```

Comment:
Though the migration of Room compiler is a non-goal, we should also migrate to `Room Gradle Plugin` in this PR.

Reference: https://developer.android.com/jetpack/androidx/releases/room